colossus/src/main/scala/colossus/protocols/http/HttpResponseParser.scala

```diff
@@ -3,68 +3,231 @@ package protocols.http
 
 import core._
 
-import akka.util.ByteString
+import akka.util.{ByteString, ByteStringBuilder}
 
 import colossus.parsing._
 import HttpParse._
 import Combinators._
 import DataSize._
+import controller._
+import service.DecodedResult
+import scala.language.higherKinds
+
+sealed trait ResponseResult 
+object ResponseResult {
+  case class StaticResponse(response: HttpResponse) extends ResponseResult
+  case class StreamResponse(sink: Sink[DataBuffer], response: HttpResponse) extends ResponseResult
+}
 
-
-object HttpResponseParser {
-
+object HttpResponseParser  {
   val DefaultMaxSize: DataSize = 10.MB
 
-  val DefaultQueueSize : Int = 100
-
-  //not totally in love with this extra object allocation, but feel its more straightforward than extractors
-  case class ParsedHeaders(headers : Seq[(String, String)]) extends HttpHeaderUtils
-
-  def static(size: DataSize = DefaultMaxSize) : HttpResponseParser[HttpResponse] = {
-    new HttpResponseParser[HttpResponse](() => maxSize(size, staticResponse))
-  }
+  //TODO: eliminate duplicate code
+  //TODO: Dechunk on static
```

Comment:
Is this comment still valid?  It looks like you are already dechunking static responses.